File: output/typescript_rest_sdk/client_sdk/models/new_pet.ts
/**
 * 
 */
export interface Newpet {
  /**
   * Name of the pet
   */
  name: String;
  /**
   * Tag associated with the pet
   */
  tag?: String;
  /**
   * URLs of pet photos
   */
  photourls?: String;
}


/**
 * Type guard to check if an object is a valid Newpet
 */
export function isNewpet(obj: any): obj is Newpet {
  if (!obj || typeof obj !== 'object') {
    return false;
  }
  
  if (!('name' in obj)) {
    return false;
  }
  
  return true;
}

/**
 * Creates a new Newpet with default values
 */
export function createNewpet(name: String, ): Newpet {
  return {
    name,
  };
}

/**
 * Validates a Newpet object
 */
export function validateNewpet(obj: Newpet): string[] {
  const errors: string[] = [];
  
  if (obj.name === undefined || obj.name === null) {
    errors.push('name is required');
  }
  
  return errors;
}

export default Newpet;

File: output/go_rest_sdk/client_sdk/models/new_pet.go
package 

import (
	"encoding/json"
	"time"
)

// Newpet 
type Newpet struct {
	// Name of the pet
Name String `json:"name"`
	// Tag associated with the pet
Tag *String `json:"tag" omitempty`
	// URLs of pet photos
Photourls *String `json:"photoUrls" omitempty`
}

// NewNewpet creates a new Newpet with required fields
func NewNewpet(name String, ) *Newpet {
	return &Newpet{
		Name: name,
	}
}

// SetTag sets the tag field
func (m *Newpet) SetTag(tag String) *Newpet {
	m.Tag = &tag
	return m
}

// GetTag returns the tag field value if set, nil otherwise
func (m *Newpet) GetTag() *String {
	return m.Tag
}

// HasTag returns true if the tag field is set
func (m *Newpet) HasTag() bool {
	return m.Tag != nil
}

// SetPhotourls sets the photoUrls field
func (m *Newpet) SetPhotourls(photourls String) *Newpet {
	m.Photourls = &photourls
	return m
}

// GetPhotourls returns the photoUrls field value if set, nil otherwise
func (m *Newpet) GetPhotourls() *String {
	return m.Photourls
}

// HasPhotourls returns true if the photoUrls field is set
func (m *Newpet) HasPhotourls() bool {
	return m.Photourls != nil
}



// Validate validates the Newpet instance
func (m *Newpet) Validate() error {
	// Add validation logic here if needed
	// Name is required
	return nil
}

// String returns a string representation of the Newpet
func (m *Newpet) String() string {
	data, _ := json.MarshalIndent(m, "", "  ")
	return string(data)
}